### ISSUE TITLE
[Feature Request] Support pem with data source tls_public_key

### ISSUE THREAD

**Issue by rukawata** [hashicorp/terraform-provider-tls#74]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
```
$ terraform -v
Terraform v0.12.23
+ provider.azurerm v2.4.0
+ provider.tls v2.1.1
```

### Affected Resource(s)
`provider.tls v2.1.1`

### Terraform Configuration Files
```hcl
data "tls_public_key" "ppk" {
  private_key_pem = "${file("my.pem")}"
}

output "ppk" {
  value = data.tls_public_key.ppk.private_key_pem
}

output "pk" {
  value = data.tls_public_key.ppk.public_key_pem
}
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
We want to use certificate generated with `azurerm_key_vault_certificate` to provision `azurerm_linux_virtual_machine`.
Currently, it's not possible to get the public key from `azurerm_key_vault_certificate` in Terraform (because azure REST API does not have that functionality), therefore we are trying to use `public_key_openssh` from data `tls_public_key`.

Of course we can use `tls_private_key` to generate a SSH key pair, but then we are not taking advantage of the azure certificate to rotate he key for example.

### Actual Behavior
```
Error: failed to decode PEM block containing private key of type "PRIVATE KEY"

  on tls.tf line 7, in data "tls_public_key" "ppk":
   7: data "tls_public_key" "ppk" {

```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-1234

**Comment by mattduguid** [comment]
same issue, need the public key to load against an azure app registration

Error: failed to decode PEM block containing private key of type "unknown"

  on .terraform/modules/***/main.tf line 82, in data "tls_public_key" "client_certificate":
  82: data "tls_public_key" "client_certificate" {

**Comment by detro** [comment]
This has been resolved in release `v3.2.1` via #182
